src/extensionsIntegrated/CSSColorPreview/main.js

```diff
@@ -319,11 +374,95 @@ define(function (require, exports, module) {
         }
     }
 
+
+    /**
+     * Responsible to update the color marks only on the modified lines
+     *
+     * @param {Editor} editor the editor instance
+     * @param {Number} fromLineNumber modification start from line number
+     * @param {Number} toLineNumber modification upto line number
+     * @return {Array.<Object>} an array of objects with all the line nos and,
+     *  the colors to be added on those lines
+     */
+    function updateColorMarks(editor, fromLineNumber, toLineNumber) {
+
+        const aColors = [];
+
+        // Match colors and push into an array
+        for (let i = fromLineNumber; i <= toLineNumber; i++) {
+            let lineText = editor.getLine(i);
+            let regx = /:[^;]*;/g;
+
+            let lineMatches = lineText.match(regx);
+            if (lineMatches) {
+                // Iterate through all matches to find valid colors
+                let validColors = [];
+                let occurrenceCount = {}; // Track occurrences for each color
+
+
+                for (let lineMatch of lineMatches) {
```

Comment:
This and code in `_getAllColorsAndLineNums` has duplication, extract the logic out to a different function.